### PR TITLE
feat(router): allow providing metadata in routes

### PR DESCRIPTION
**Pull request by devknoll**
The Meta service directly exposes HTMLMetaElements, and therefore requires DOM emulation to use. In order to allow setting page metadata without DOM emulation, introduce support on the route, similar to title.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by devknoll** [comment]
friendly ping @AndrewKushnir

File: packages/router/src/models.ts [hunk 1]
```diff
@@ -331,6 +331,23 @@ export type RunGuardsAndResolvers =
   | 'always'
   | ((from: ActivatedRouteSnapshot, to: ActivatedRouteSnapshot) => boolean);
 
+/**
+ * Represents metadata for a particular route.
+ *
+ * @publicApi
+ */
+export interface MetadataEntry {
```
Comment (dgp1130) [inline packages/router/src/models.ts:339]: Drive by: Doesn't TS' existing DOM APIs have this information? Would we use one of those types?

That might only contain properties and not attributes, but worth taking a look just to be sure.

Comment (devknoll) [reply packages/router/src/models.ts:339]: Worth looking into. This definition came from the `Meta` service.

**Comment by atscott** [comment]
How common will it be to define metadata on routes? I'm wondering if the `MetadataStrategy` should be provided with a feature opt-in rather than having it there by default and not tree-shakeable. We can add a dev-mode check to the [config validation](https://github.com/angular/angular/blob/main/packages/router/src/utils/config.ts#L61) that throws if the feature isn't provided but the `metadata` property is present.

**Pull request closed without merging**